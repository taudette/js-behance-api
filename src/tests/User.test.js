import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import UserContainer from '../containers/User'
import UserProfileComponent from '../components/UserProfile'
import UserProjectsComponent from '../components/UserProjects'
import UserExperienceComponent from '../components/UserExperience'
import FollowersComponent from '../components/Followers'
import FollowingComponent from '../components/Following'

configure({ adapter: new Adapter() });

describe('User container', () => {
  it('UserContainer is connected', () => {
    const mockStore = configureStore([]);
    const store = mockStore({});
    const wrapper = shallow(<Provider store={store}>
      <UserContainer />
    </Provider>);
    expect(wrapper.text()).toBe('<Connect(UserContainer) />');
  })
})

describe('UserProfile component', () => {
	const userProfile = shallow(<UserProfileComponent />);
	it('User profile component renders profile correctly', () => {
		expect(userProfile).toMatchSnapshot();
	});
})

describe('UserProjects component', () => {
	const userProjects = shallow(<UserProjectsComponent />);
	it('User projects component renders projects correctly', () => {
		expect(userProjects).toMatchSnapshot();
	});
})

describe('Followers component', () => {
	const followers = shallow(<FollowersComponent />);
	it('Followers component renders projects correctly', () => {
		expect(followers).toMatchSnapshot();
	});
})

describe('Following component', () => {
	const following = shallow(<FollowingComponent />);
	it('Following component renders projects correctly', () => {
		expect(following).toMatchSnapshot();
	});
})