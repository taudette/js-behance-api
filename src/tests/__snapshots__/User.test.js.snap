// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Followers component Followers component renders projects correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FollowersComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Following component Following component renders projects correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FollowingComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`UserProfile component User profile component renders profile correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserProfileComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "panel",
      "bsStyle": "default",
      "children": undefined,
      "className": "Profile-panel",
      "defaultExpanded": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "panel",
        "bsStyle": "default",
        "children": undefined,
        "className": "Profile-panel",
        "defaultExpanded": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`UserProjects component User projects component renders projects correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserProjectsComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "panel",
      "bsStyle": "default",
      "children": Array [
        <h3>
          Projects
</h3>,
        <Row
          bsClass="row"
          componentClass="div"
>
          <p>
                    No projects are currently available
          </p>
</Row>,
      ],
      "defaultExpanded": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Projects",
        },
        "ref": null,
        "rendered": "Projects",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "row",
          "children": <p>
            No projects are currently available
</p>,
          "componentClass": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No projects are currently available",
          },
          "ref": null,
          "rendered": "No projects are currently available",
          "type": "p",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "panel",
        "bsStyle": "default",
        "children": Array [
          <h3>
            Projects
</h3>,
          <Row
            bsClass="row"
            componentClass="div"
>
            <p>
                        No projects are currently available
            </p>
</Row>,
        ],
        "defaultExpanded": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Projects",
          },
          "ref": null,
          "rendered": "Projects",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "row",
            "children": <p>
              No projects are currently available
</p>,
            "componentClass": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No projects are currently available",
            },
            "ref": null,
            "rendered": "No projects are currently available",
            "type": "p",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
